oRouter.js

```diff
@@ -67,20 +70,79 @@ class oRouter {
         );
     }
 
-    static addSearchParams(params){
-        let paramsStr = '';
-        Object.entries(params).forEach(([key, value]) => {
-            paramsStr += `${encodeURIComponent(key)}=${encodeURIComponent(value)}`
+    static setSearchParam(key, value){
+        if(!key || typeof key !== 'string' || typeof value === 'object') return false;
+        return oRouter.setSearchParams({
+            [key]: String(value)
         });
+    }
+
+    static unsetSearchParam(key){
+        delete oRouter.#routingParams.searchParams[key];
+
         const { url } = oRouter.#routingParams;
+        let paramsStr = oRouter.#searchParamsToString();
 
-        url.search = url.search.length ?
-            `${url.search}&${paramsStr}` :
-            `?${paramsStr}`;
+        url.search = `?${paramsStr}`;
+        return oRouter.#changeState(url);
+    }
 
-        oRouter.#changeState(url);
+    static setSearchParams(params){
+        if(typeof params !== 'object') return false;
+        if(!Object.keys(params).length) return true;
```

Comment:
Nie mogę dać suggestion, ale może zamiast robić dwa ify to może zrobić jeden
`(!Object.keys(params).length && params.constructor === Object ? true : false`